src/main/java/edu/ucsb/cs156/organic/controllers/CoursesController.java

```diff
@@ -213,5 +201,24 @@ public Iterable<Staff> getStaff(
         Iterable<Staff> courseStaff = courseStaffRepository.findByCourseId(course.getId());
         return courseStaff;
     }
-  
+
+    //  There is a GET endpoint /api/course/get?id=123 that gets the course with id 123 if you are an admin, or if you are logged in and on the course staff.
+    @Operation(summary= "Get a single course by id")
+    @PreAuthorize("hasAnyRole('ROLE_USER', 'ROLE_ADMIN')")
```

Comment:
I think with `hasAnyRole` you have to drop the `ROLE_` prefix, like this (I could be mistaken).

I realize this seems inconsistent, but that's apparently how the Spring folks did it.

```suggestion
    @PreAuthorize("hasAnyRole('USER', 'ADMIN')")
```

See: https://www.baeldung.com/spring-security-expressions